examples/with-supertokens/middleware.ts

```diff
@@ -0,0 +1,36 @@
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+import { SessionContainer } from 'supertokens-node/recipe/session'
+import { ensureSuperTokensInit } from './config/backendConfig'
+import { withSession } from './app/sessionHandlers';
+
+ensureSuperTokensInit();
+
+export async function middleware(
+  request: NextRequest & { session?: SessionContainer }
+) {
+  if (request.headers.has("x-user-id")) {
+    console.warn("The FE tried to pass x-user-id, which is only supposed to be a backend internal header. Ignoring.");
+    request.headers.delete("x-user-id");
+  }
+
+  if (request.nextUrl.pathname.startsWith('/api/auth')) {
+    // this hits our pages/api/auth/* endpoints
+    return NextResponse.next()
+  }
+
+  return withSession(request, async (session) => {
```

Comment:
@nkshah2 said that maybe you can't return a response object directly (he is not sure), but we are doing this in case the session's access token is expired. Does that work for sure?

Reply:
It does, just tested it.